saviSimulation/savi/jason_processing/UxVBehavior.java

```diff
@@ -231,16 +245,17 @@ protected boolean isSensorFailing (double probability) {
 	}
 
 	
-	// generate random value for a normal distribution (mean, stdDev)
+	// Generate random value for a normal distribution (mean, stdDev)
+	// nextGaussian returns a value for the normal distribution (0,1)
+	// multiply for the stdDev to get error and this error value is added to the mean
 	protected double calculateFailureValue (double mean, double stdDev) {
 		return ((rand.nextGaussian()*stdDev)+mean);
 	}
-
 	
-	public static void setSeed(int seed) {
-		if(seed != -1) {
-			rand = new Random(seed);
-		}
+	
+	
+	public static void setSeed(Random ran) {
+			rand = ran;		
```

Comment:
Should do a null check here. Also, any time that rand is updated, use this setter.